tests/arch/arm/arm_thread_swap/src/arm_thread_arch.c

```diff
@@ -431,6 +432,18 @@ void test_arm_thread_swap(void)
 		: "memory"
 	);
 
+	#if !defined(CONFIG_NO_OPTIMIZATIONS)
+	__asm__ volatile (
+		"mov r0, %1;\n\t"
+		"stm %0, {r0};\n\t"
+		:
+		: "r" (&ztest_swap_return_val), "r" (swap_return_val)
+		: "memory"
```

Comment:
I believe (verified assembly, tested sanitycheck and nucleo_l476rg`) that this can be reduced to:
```
	__asm__ volatile (
		"stm %0, {%1};\n\t"
		:
		: "r" (&ztest_swap_return_val), "r" (swap_return_val)
		: "memory"
	);
```
which avoids an unnecessary `mov r0, r0`.

Reply:
Yeap, I agree with simplification. Fixed